### DISCUSSION TITLE
picture with an esp01f and ufs

### DISCUSSION THREAD

**Discussion by DennisXK** [arendst/Tasmota#22112] [Q&A]
Hello, I would like to display a logo/small image on the web UI on an esp01f. With a Wemos D1 mini I managed it successfully with the UFS.

**Comment by sfromis** [comment]
Not a feature of Tasmota. You could replace the flash chip to 4MB, giving you the opportunity of having a file system. Or you could host the image elsewhere, like another Tasmota device.

Another approach could be to base64 encode the image, and put the resulting string into a reference like:
`<img src="data:image/png;base64,xxxxxx"/>`

**Reply by DennisXK** [reply]
I'll take a look at the base64 version. Thanks for the tip

**Comment by DennisXK** [comment]
I tried it out, unfortunately the Base64 code is extremely large. So I can't get it integrated into Tasmota Script.

**Reply by sfromis** [reply]
An actually small logo should be doable, like the Tasmota logo:
`<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeAQMAAAAB/jzhAAAABlBMVEUAAAD///+l2Z/dAAAAbklEQVQI13XOsQmAMBQE0NvABQQXCbpYwAUCWUCcxULIEnYRLI3YRPjkewqWXvGq4zioqkDlIfckdSQ2JHjiA2mioHSHQPosyCoWR2oN4lTVCC44eHPOGO21otiLPeaft1IM97Tmsg4bWfbvBrkB+LxlR7I/9NcAAAAASUVORK5CYII="/>`

**Reply by DennisXK** [reply]
ok 30x30px is too small again. I was thinking 200x200px. But when I paste your code and >W I don't see the logo.
`>W
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeAQMAAAAB/jzhAAAABlBMVEUAAAD///+l2Z/dAAAAbklEQVQI13XOsQmAMBQE0NvABQQXCbpYwAUCWUCcxULIEnYRLI3YRPjkewqWXvGq4zioqkDlIfckdSQ2JHjiA2mioHSHQPosyCoWR2oN4lTVCC44eHPOGO21otiLPeaft1IM97Tmsg4bWfbvBrkB+LxlR7I/9NcAAAAASUVORK5CYII="/>`

**Reply by sfromis** [reply]
Such a big image is **way** outside of what you said about a small image/logo :wink: 

Not doing things with scripting, IDK if your test case "should" work, but when I have that image source in a small web page, and display it in a browser, I see the logo as expected.
